pkg/apis/eventing/v1alpha1/trigger_types.go

```diff
@@ -58,9 +58,19 @@ type TriggerSpec struct {
 	// +optional
 	DeprecatedGeneration int64 `json:"generation,omitempty"`
 
-	Broker     string          `json:"broker,omitempty"`
+	Broker string `json:"broker,omitempty"`
+
+	// +optional
+	Filter *TriggerFilter `json:"filter,omitempty"`
+
 	Subscriber *SubscriberSpec `json:"subscriber,omitempty"`
+}
+
+type TriggerFilter struct {
+	ExactMatch TriggerFilterAttributes `json:"exactMatch,omitempty"`
```

Comment:
For future oneOf reasons, I think this should be a pointer. That way it is easier to distinguish exactly one TriggerFilter field was set.

Note that this also implies we need to do something in the webhook to default and/or validate it.

Reply:
Changed it to pointer. Added another validation, check if makes sense.